frontend-v2/src/app/course/[id]/page.tsx

```diff
@@ -0,0 +1,149 @@
+import Image from "next/image";
+import waves from "../../../assets/waves.svg";
+import { LinkIcon } from "@heroicons/react/24/solid";
+import { Suspense } from "react";
+import TermsGroup from "@/components/TermsGroup";
+import Link from "next/link";
+import DoughnutChart from "@/components/DoughnutChart";
+import { notFound } from "next/navigation";
+import ReviewsBar from "../../../components/review/ReviewsBar";
+import Rating from "@/components/Rating";
+import ReviewSearchbar from "@/components/review/ReviewSearchBar";
+
+/**
+ * Fetch course from backend based on the given id
+ */
+const getCourse = async (id: string) => {
+  const response = await fetch(`http://localhost:3030/api/v1/courses`, {
+    method: "GET",
+  });
+
+  // Return undefined if status code is not 200
+  if (response.status !== 200) return;
+
+  // Return course
+  const result = await response.json();
+  return result.courses.find(
+    (course: any) => course.courseCode === id.toUpperCase()
+  );
+};
+
+/**
+ * Fetch reviews from backend based on the given id
+ */
+const getReviews = async (id: string) => {
```

Comment:
I think we should abstract these functions outside of the component and have a common fetcher that we can reuse, since these can get complicated:
1. we need to change URL depending on the environment (local vs prod vs staging)
2. need to prevent next js caching
3. auth for some apis